cloudferrylib/os/network/neutron.py

```diff
@@ -1345,11 +1353,12 @@ def get_resource_hash(neutron_resource, *args):
 
     def get_new_extnet_id(self, src_net_id, src_nets, dst_nets):
         if src_net_id in self.ext_net_map:
-            dst_net_id = self.ext_net_map[src_net_id]
+            return self.ext_net_map[src_net_id]
         else:
             net_hash = self.get_res_hash_by_id(src_nets, src_net_id)
-            dst_net_id = self.get_res_by_hash(dst_nets, net_hash)['id']
-        return dst_net_id
+            dst_net = self.get_res_by_hash(dst_nets, net_hash)
+            if dst_net:
```

Comment:
This only hides the problem, not resolves it. The real problem is that network with the given hash does not exist on destination, and this error must not occur at all.